eox_tagging/api/v1/filters.py

```diff
@@ -25,38 +25,28 @@ class Meta:  # pylint: disable=old-style-class
         model = Tag
         fields = ['key', 'created_at', 'activated_at', 'status', 'course_id', 'enrolled', 'enrollments', 'username']
 
-    def filter_course_id(self, queryset, name, value):  # pylint: disable=unused-argument
-        """Filter that returns the tags associated with course_id."""
+    def filter_by_target_object(self, queryset, name, value):
+        """Filter that returns the tags associated with target."""
+        TARGET_TYPES = {
+            "course_id": "courseoverview",
+            "username": "user",
+            "enrollment": "courseenrollment",
+        }
         if value:
-            try:
-                queryset = Tag.objects.find_all_tags_for(target_type="CourseOverview",
-                                                         target_id={"course_id": str(value)})
-            except Exception:  # pylint: disable=broad-except
-                return queryset.none()
-
-        return queryset
-
-    def filter_username(self, queryset, name, value):  # pylint: disable=unused-argument
-        """Filter that returns the tags associated with username."""
-        if value:
-            try:
-                queryset = Tag.objects.find_all_tags_for(target_type="user",
-                                                         target_id={"username": str(value)})
-            except Exception:  # pylint: disable=broad-except
-                return queryset.none()
-
-        return queryset
-
-    def filter_enrolled(self, queryset, name, value):  # pylint: disable=unused-argument
-        """Filter that returns tags in which the target is a course where the user is enrolled in."""
-        if value:
-            enrollment = {  # pylint: disable=broad-except
+            TARGET_IDENTIFICATION = {}
+            TARGET_IDENTIFICATION["enrollment"] = {
                 "username": self.request.user.username,
-                "course_id": str(value)
+                "course_id": str(value),
+            }
+            DEFAULT = {
+                name: str(value),
             }
             try:
-                queryset = Tag.objects.find_all_tags_for(target_type="courseenrollment",
-                                                         target_id=enrollment)
+                filter_params = {
+                    "target_type": TARGET_TYPES.get(name),
```

Comment:
are you testing when this returns None ?

Reply:
name is an attribute of CharFilter, I only call filter_by_target_object with names that are defined so it shouldn't be None. Let me see how can I generate the test where name is None

Reply:
it is worth testing it? Considering that the filter is not being called isolated @andrey-canon

Reply:
I mean when the name is not in the dictionary,  for example what happen when name is `patito`

Reply:
Then it won't find any matching target and it will return an empty queryset. I'll add some tests for that